### PR TITLE
[LED-116] Last DB file Load

### PR DESCRIPTION
**Pull request by xxshadowfang**
This uses java preferences to load the previously used db file. 

This needs tested on a Mac.

### CODE REVIEW

File: src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java [hunk 1]
```diff
@@ -112,8 +122,13 @@ private void login() {
         try {
             DbController.INSTANCE.initialize(this.filePath, pwd);
             Startup.INSTANCE.newStage(new Scene(new MainPageController()), "Ledger");
+            Preferences preferences = Preferences.userRoot().node("LedgerSoftware");
+            preferences.put(LAST_DATABASE_FILE_KEY,this.filePath);
+            preferences.flush();
```
Comment (shellajt) [inline src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java]: Don't you want this line moved up a couple?

File: src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java [hunk 2]
```diff
@@ -112,8 +122,13 @@ private void login() {
...
         } catch (StorageException e) {
             this.setupErrorPopup("Unable to connect to database", e);
+        } catch (BackingStoreException e){
+
```
Comment (shellajt) [inline src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java]: Empty catch blocks are bad...

**Comment by xxshadowfang** [comment]
Sorry I tested both cases of Exiting the mainpage and logging out on linux and they worked but on windows if you exited instead of logging out the db filed didnt persist. I dont know if it exists on the mac. 

Moving time of the execution makes the filepath persist in windows if you exit out of the mainpage. Also i added an error popup if an exception is caused by the preferences on persist.

File: src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java [hunk 3]
```diff
@@ -111,9 +121,16 @@ private void login() {
...
+            Preferences preferences = Preferences.userRoot().node("LedgerSoftware");
+            preferences.put(LAST_DATABASE_FILE_KEY,this.filePath);
+            preferences.flush();
             Startup.INSTANCE.newStage(new Scene(new MainPageController()), "Ledger");
+
+
         } catch (StorageException e) {
             this.setupErrorPopup("Unable to connect to database", e);
+        } catch (BackingStoreException e){
+            this.setupErrorPopup("Unable store last Db file", e);
```
Comment (shellajt) [inline src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java]: I don't know if the user cares about this error. Might be more appropriate in System.err

**Pull request merged**